### ISSUE TITLE
Does PPSSPP exit gracefully from any application

### ISSUE THREAD

**Issue by tilkinsc** [hrydgard/ppsspp#11042]
I am working with some homebrew and it is typical for a callback handler which suspends a kernel thread to keep the program and perchance the while loop active. Is there a way in PPSSPP to resume such a thread so exiting can gracefully go to the PPSSPP XMB? (For example, HOME -> EXIT on physical PSP)

**Comment by hrydgard** [comment]
Does it not work to simply call sceKernelExitGame(), or does that hang or something?

**Comment by tilkinsc** [comment]
I guess I am more worried about crashing the game to close it with the savestate menu thing. sceKernelExitGame() does indeed work, but I also wanted to make sure that my code is being executed, too. For example, saving data before closing.

**Comment by unknownbrackets** [comment]
I don't think we actually call the exit callback, ever.  When the user selects exit game we abort emulation currently.  Maybe we should, although that would require a timeout or something in case the game doesn't actually quit, I suppose.

-[Unknown]

**Comment by hrydgard** [comment]
Since "real" games never call sceKernelExitGame, this has indeed not been prioritized. Marking as feature request.
